### PR TITLE
feature/qp-100-add-inventory-changed-eventing.

### PR DESCRIPTION
**Pull request by LoganGore**
Add inventory changed eventing that notifies client with diff of inventory.

### CODE REVIEW

File: src/AssistantAPIClient.js [hunk 1]
```diff
@@ -136,6 +141,34 @@ class AssistantAPIClient{
   deleteKind = input => APICall('deleteKind', input)
 
   getKindById = id => APICall('getKindById', id)
+
+  //
+  // Inventory
+  //
+
+  // Called when adding inventory changed listener.
```
Comment (bzvestey) [inline src/AssistantAPIClient.js]: `Called when adding an inventory changed event listener.`

File: src/AssistantAPIClient.js [hunk 2]
```diff
@@ -136,6 +141,34 @@ class AssistantAPIClient{
...
+  enableInventoryChangedNotification = async () => APICall('enableInventoryChangedNotification')
+
+  // Called when removing inventory changed listener if listener count is zero.
```
Comment (bzvestey) [inline src/AssistantAPIClient.js]: `Called when removing the last inventory changed listener`

File: src/AssistantAPIClient.js [hunk 3]
```diff
@@ -136,6 +141,34 @@ class AssistantAPIClient{
...
+  disableInventoryChangedNotification = async () => APICall('disableInventoryChangedNotification')
+
+  addInventoryChangedListener = async cb => {
+    this.enableInventoryChangedNotification()
+    EventEmitter.addListener('inventoryChanged', cb)
+  }
+
+  removeInventoryChangedListener = async cb => {
+    // If the callback is not provided, then remove all listeners.
```
Comment (bzvestey) [inline src/AssistantAPIClient.js]: `If the callback is not provided, then remove all of the listeners.`

File: src/AssistantAPIClient.js [hunk 4]
```diff
@@ -136,6 +141,34 @@ class AssistantAPIClient{
...
+    if(cb){
+      EventEmitter.removeListener('inventoryChanged', cb)
+    } else{
+      EventEmitter.removeAllListeners('inventoryChanged')
+    }
+    // After removal, determine if we should implicitly disable notification.
```
Comment (bzvestey) [inline src/AssistantAPIClient.js]: `After removing the callback, determine if the notifications should be implicitly disabled.`

File: src/AssistantAPIClient.js (hunk 3 above)
Comment (LoganGore) [reply src/AssistantAPIClient.js]: Either should be correct in modern English.

Comment (LoganGore) [reply src/AssistantAPIClient.js]: In this case, I'll make an exception add it only because the other comments follow this same style.

Comment (bzvestey) [reply src/AssistantAPIClient.js]: This way just flows better to me.  It is not just about be a valid sentence, but trying to get a sentence that flows well is also nice.

**Pull request merged**